### ISSUE TITLE
Acartia ssp and Oithona spp not grouped/summed?

### ISSUE THREAD

**Issue by LennertSchepers** [vlizBE/BlueCloud-zooplankton#1]
Hi @salvafern 

I think the different Acartia spp and Oithona spp occurrences are not summed at the moment?
(you only changed the name to Acartia, but did not sum if they other spp were present in the same sample?)

**Comment by salvafern** [comment]
Hi @LennertSchepers 

Thanks for the feedback. 

After increasing the taxon rank of the _Acartia_ species to its genus I recalculated the number of individuals in invidualCount in each row as the sum of all the counts per specific eventID and aphiaID (it would also work using the scientificName or scientificNameID). The following distinct() removed the duplicates, therefore obtaining one row per eventID and scientificName, indicating the sum of individualCounts. This approach is the R equivalent to the [SQL Window functions.](https://www.sqltutorial.org/sql-window-functions/sql-partition-by/)

Note this is possible because of [removing the occurrenceID in line 15](https://github.com/vlizBE/BlueCloud-zooplankton/commit/7bc174a4a014b1ce785715c22355f28655cef06d#diff-2a56637f9a30acf79cbfcebf038273faL15).

Please try the following code for visualization, [from line 56](https://github.com/vlizBE/BlueCloud-zooplankton/blob/7bc174a4a014b1ce785715c22355f28655cef06d/R/read_cpr.R#L56):

```r
# Low Acartia and Oithona species taxonomy to genus
# data$scientificName[grep("Acartia", data$scientificName, ignore.case = TRUE)] <- "Acartia"
data$scientificNameID[grep("Acartia", data$scientificName, ignore.case = TRUE)] <- "urn:lsid:marinespecies.org:taxname:104108"
data$aphiaID[grep("Acartia", data$scientificName, ignore.case = TRUE)] <- 104108

# data$scientificName[grep("Oithona", data$scientificName, ignore.case = TRUE)] <- "Oithona"
data$scientificNameID[grep("Oithona", data$scientificName, ignore.case = TRUE)] <- "urn:lsid:marinespecies.org:taxname:106485"
data$aphiaID[grep("Oithona", data$scientificName, ignore.case = TRUE)] <- 106485

# Aggregate counts using dplyr v 0.8.3
library(dplyr)
packageVersion("dplyr")

# View data before distinct
data %>% mutate(individualCount = as.integer(individualCount)
  ) %>% group_by(id, aphiaID
  ) %>% mutate(individualCount_sum = sum(individualCount) 
  ) %>% select(id, scientificName, aphiaID, individualCount, individualCount_sum
  ) %>% filter(individualCount != individualCount_sum
  ) %>% arrange(id
  ) %>% View()

# View data after distinct
data %>% mutate(individualCount = as.integer(individualCount)
  ) %>% group_by(id, aphiaID
  ) %>% mutate(individualCount_sum = sum(individualCount) 
  ) %>% select(id, scientificName, aphiaID, individualCount, individualCount_sum
  ) %>% filter(individualCount != individualCount_sum
  ) %>% arrange(id
  ) %>% select(-individualCount, -scientificName
  ) %>% distinct(
  ) %>% View()
```

Note that for visualization purposes, I set individualCount not equal to individualCount_sum. In some cases this produces that the sum doesn't seem to match. This happens typically cause you are seeing one absence (individualCount = 0). In the same event, there was only this absence and another occurrence, therefore the new individualCount_sum is the same as the second occurrence. Check for example id == '124AT-24'.

**Comment by LennertSchepers** [comment]
Hi @salvafern, can you make a pull request with the changes above? thanks!

**Comment by LennertSchepers** [comment]
solved by commit 7bc174a4a014b1ce785715c22355f28655cef06d
